plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/settings/CodeWhispererConfigurable.kt

```diff
@@ -87,6 +89,72 @@ class CodeWhispererConfigurable(private val project: Project) :
+                            override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                enabled(isCodeWhispererEnabled(project))
+                            }
+                        }
+                    )
+                    enabled(invoke)
+                    bindSelected(codeWhispererSettings::isProjectContextEnabled, codeWhispererSettings::toggleProjectContextEnabled)
+                }.comment(message("aws.settings.codewhisperer.project_context.tooltip"))
+            }
+
+            row(message("aws.settings.codewhisperer.project_context_index_thread")) {
+                intTextField(
+                    range = IntRange(0, 50)
+                ).bindIntText(codeWhispererSettings::getProjectContextIndexThreadCount, codeWhispererSettings::setProjectContextIndexThreadCount)
+                    .align(AlignX.FILL).apply {
+                        connect.subscribe(
+                            ToolkitConnectionManagerListener.TOPIC,
+                            object : ToolkitConnectionManagerListener {
+                                override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                    enabled(isCodeWhispererEnabled(project))
+                                }
+                            }
+                        )
+                        enabled(invoke)
+                    }.comment(message("aws.settings.codewhisperer.project_context_index_thread.tooltip"))
+            }
+
+            row(message("aws.settings.codewhisperer.project_context_index_max_size")) {
+                intTextField(
+                    range = IntRange(1, 250)
+                ).bindIntText(codeWhispererSettings::getProjectContextIndexMaxSize, codeWhispererSettings::setProjectContextIndexMaxSize)
+                    .align(AlignX.FILL).apply {
+                        connect.subscribe(
+                            ToolkitConnectionManagerListener.TOPIC,
+                            object : ToolkitConnectionManagerListener {
+                                override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                    enabled(isCodeWhispererEnabled(project))
+                                }
+                            }
+                        )
+                        enabled(invoke)
+                    }.comment(message("aws.settings.codewhisperer.project_context_index_max_size.tooltip"))
+            }
+
+            row {
+                checkBox(message("aws.settings.codewhisperer.project_context_gpu")).apply {
+                    connect.subscribe(
+                        ToolkitConnectionManagerListener.TOPIC,
+                        object : ToolkitConnectionManagerListener {
+                            override fun activeConnectionChanged(newConnection: ToolkitConnection?) {
+                                enabled(isCodeWhispererEnabled(project))
```

Comment:
zoe actually let's use this one instead of `isCWEnabled()`
```suggestion
                                enabled(ToolkitConnectionManager.getInstance(project).activeConnectionFeatureFor(QConnection.getInstance()))
```

Reply:
i wanna deprecate that stale func

Reply:
should we keep it for now and change all of them at once?

Reply:
ok that works for me